src/compas_timber/ghpython/components/CT_Bake_BoxMap/code.py

```diff
@@ -14,44 +16,47 @@
 
 
 class BakeBoxMap(component):
-    def RunScript(self, model, map_size, bake):
-        if map_size and len(map_size) != 3:
-            self.AddRuntimeMessage(
-                Error, "Input parameter MapSize requires exactly three float values (scale factors in x,y,z directions)"
-            )
-            return
-
-        if map_size:
-            dimx, dimy, dimz = map_size
-        else:
-            # for the pine 251 material bitmap, rotated
-            dimx = 0.2
-            dimy = 0.2
-            dimz = 1.0
+    def RunScript(self, model, beam_map_size, plate_map_size, beam_layer_name, plate_layer_name, swap_uv, bake):
+        def get_dimensions(map_size, default):
+            return map_size if map_size else default
 
+        # Check if model exists and baking is enabled
         if not model:
             self.AddRuntimeMessage(Warning, "Input parameters Model failed to collect any Beam objects.")
             return
 
-        if not bake:
+        if not bake:  # Exit early if bake is False
             return
 
+        # Save current Grasshopper document context
+        ghdoc = sc.doc
+
+        # Switch document context to Rhino's active document for baking
+        sc.doc = Rhino.RhinoDoc.ActiveDoc
+
+        # Define dimensions
+        b_dimx, b_dimy, b_dimz = get_dimensions(beam_map_size, [0.2, 0.2, 1.0])
+        p_dimx, p_dimy, p_dimz = get_dimensions(plate_map_size, [1.0, 1.0, 1.0])
+
         try:
-            frames = [frame_to_rhino(b.frame) for b in model.beams]
-            breps = [beam.geometry.native_brep for beam in model.beams]
+            # Bake beams
+            beam_frames = [frame_to_rhino(b.frame) for b in model.beams]
+            beam_breps = [beam.geometry.native_brep for beam in model.beams]
+            self.add_brep_to_document(beam_breps, beam_frames, beam_layer_name, b_dimx, b_dimy, b_dimz, False)
 
-            if frames and breps:
-                rs.EnableRedraw(False)
+            # Bake plates
+            plate_frames = [frame_to_rhino(p.frame) for p in model.plates]
+            plate_breps = [plate.geometry.native_brep for plate in model.plates]
+            self.add_brep_to_document(plate_breps, plate_frames, plate_layer_name, p_dimx, p_dimy, p_dimz, swap_uv)
 
-                for brep, frame in zip(breps, frames):
-                    guid = ActiveDoc.Objects.Add(brep)
-                    boxmap = self.create_box_map(frame, dimx, dimy, dimz)
-                    ActiveDoc.Objects.ModifyTextureMapping(guid, 1, boxmap)
         finally:
             rs.EnableRedraw(True)
 
+            # Restore document context back to Grasshopper
+            sc.doc = ghdoc
```

Comment:
Although I'm sure `rs.EnableRedraw()` shouldn't cause any issues, I would do `sc.doc=ghdoc` before. Just to be safe.